corehq/apps/users/models.py

```diff
@@ -830,6 +830,13 @@ def check_password(self, password):
         dummy.password = self.password
         return dummy.check_password(password)
 
+    def invoke_superuser(self):
+        return self.is_superuser
+
+
+# monkey patch
+User.invoke_superuser = lambda self: self.is_superuser
```

Comment:
My original implementation operated on the request instead of the user, and it turned out to be sort of untenable because of all the places `.is_superuser` is used deep into things. There's also the chance that it would be used in a context in which there is no request, like a celery task, so while those might require individualized attention to get right, it seemed like operating on the request was setting it up to be much harder than it needed to be.

Reply:
I'm open to making it `x(user)` instead of `user.x()` (where `x` is whatever we decide to call it). I chose `user.x()` because it seemed like the simpler change from the caller's perspective, but at this point I think I agree with you that `x(user)` would be the less surprising/magical of the two.